src/Workspaces/Core/Portable/EmbeddedLanguages/LanguageServices/LanguageCommentDetector.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.LanguageServices
+{
+    /// <summary>
+    /// Helps match patterns of the form: language=name,option1,option2,option3
+    /// 
+    /// All matching is case insensitive, with spaces allowed between the punctuation. Option values will be or'ed
+    /// together to produce final options value.  If an unknown option is encountered, processing will stop with
+    /// whatever value has accumulated so far.
+    /// 
+    /// Option names are the values from the TOptions enum.
+    /// </summary>
+    internal struct LanguageCommentDetector<TOptions> where TOptions : struct, Enum
+    {
+        private static readonly Dictionary<string, TOptions> s_nameToOption =
```

Comment:
The assumption here is TOptions is a flags enum? Does something need to enforce that?

Reply:
sure.  i can add that in followup.